app/src/main/java/com/sapuseven/untis/receivers/AutoMuteSetup.kt

```diff
@@ -40,7 +40,9 @@ class AutoMuteSetup : LessonEventSetup() {
 				alarmManager.setExact(AlarmManager.RTC_WAKEUP, item.startDateTime.millis, pendingMuteIntent)
 				Log.d("AutoMuteSetup", "${item.periodData.getShortTitle()} mute scheduled for ${item.startDateTime}")
 
-				if (item.endDateTime == it.second?.startDateTime) return@forEach // No break exists, don't unmute
+				val minimumBreakLength = PreferenceUtils.getPrefInt(preferenceManager, "preference_automute_minimum_break_length").toLong()
+				val endDateTimeWithMinimumBreakLength = item.endDateTime.withDurationAdded(minimumBreakLength, 1000 * 60)
+				if (it.second != null && endDateTimeWithMinimumBreakLength >= it.second!!.startDateTime) return@forEach // No break exists or break is too short, don't unmute
```

Comment:
I'd actually suggest to just check the break length directly, it seems a bit simpler to me.
Note that I didn't test this.

```suggestion
				val minimumBreakLengthMillis = PreferenceUtils.getPrefInt(preferenceManager, "preference_automute_minimum_break_length") * 60 * 1000
				if (it.second != null && it.second!!.startDateTime.millisOfDay - item.endDateTime.millisOfDay < minimumBreakLengthMillis)
					return@forEach // No break exists or it's too short, don't unmute
```

Reply:
I think this could be an issue if the two lessons are on different days but this normally shouldn't be the case

Reply:
The items are always on the same day because only the current day is requested from the Untis API. As long as the API doesn't change, this shouldn't be a problem.